### PR TITLE
[AGW] pipelined: uplink_br: handle ingress vlan packets

### PR DESCRIPTION
**Pull request by pshelar**
## Summary

Due to OVS issue in handling ingress vlan packets,
this patch introduce veth pair to get rid of vlan header
on ingress. This could be removed once OVS is fixed.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

## Additional Information
Verified datapath manually on magma dev.

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/python/magma/pipelined/app/uplink_bridge.py [hunk 1]
```diff
@@ -202,3 +245,16 @@ def _set_sgi_static_ip(self):
             self.logger.debug("SGi ip address config: [%s]", set_ip_cmd)
         except subprocess.SubprocessError as e:
             self.logger.warning("Error while setting SGi IP: %s", e)
+
+    def _setup_vlan_pop_dev(self):
```
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/app/uplink_bridge.py:250]: Wait this is only used in testing? Lets move it to the test file then, no reason to do this for production datapath

File: lte/gateway/python/magma/pipelined/app/uplink_bridge.py [hunk 2]
```diff
@@ -29,12 +30,15 @@ class UplinkBridgeController(MagmaController):
     UPLINK_OVS_BRIDGE_NAME = 'uplink_br0'
     DEFAULT_UPLINK_PORT_NANE = 'eth3'
     DEFAULT_UPLINK_MAC = '11:22:33:44:55:66'
+    DEV_VLAN_IN = 'vlan_pop_in'
```
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/app/uplink_bridge.py]: mbe DEFAULT_DEV_VLAN_IN

File: lte/gateway/python/magma/pipelined/app/uplink_bridge.py [hunk 3]
```diff
@@ -94,12 +105,44 @@ def initialize_on_connect(self, datapath):
...
+                     self.config.virtual_mac)
+            actions = "output:%s" % self.config.uplink_patch
+            self._install_flow(1000, match, actions)
+        else:
+            # 2.b. All ingress IP traffic for UE mac
+            match = "in_port=%s,ip, dl_dst=%s" % \
+                    (self.config.uplink_eth_port_name,
+                     self.config.virtual_mac)
+            actions = "output:%s" % self.config.uplink_patch
+            self._install_flow(1000, match, actions)
```
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/app/uplink_bridge.py]: Seems like a lot of magic numbers, pull out 1000/500/100 into global const priority definitions?

**Review by koolzz — CHANGES_REQUESTED** [review]

File: lte/gateway/python/magma/pipelined/app/uplink_bridge.py (hunk 1 above)
Comment (pshelar) [reply lte/gateway/python/magma/pipelined/app/uplink_bridge.py:250]: the port number are used for deterministic test.

**Review by koolzz — APPROVED** [review]

**Pull request merged**